### ISSUE TITLE
feature request - functions to get and set initial values

### ISSUE THREAD

**Issue by drjhw** [paul-buerkner/brms#883]
brms provides the helpful functions get_prior() and set_prior() to allow inspection and setting of priors

Could you kindly provide analogous functions for inspecting and setting initial values?

**Comment by paul-buerkner** [comment]
Initial values are set by (r)stan automatically. You can set them yourself
by using the inits argument.

drjhw <notifications@github.com> schrieb am Fr., 24. Apr. 2020, 11:02:

**Comment by drjhw** [comment]
Yes, it is easy to use inits to set a single value. However, it is not clear how to set specific values for specific parameters.
There has already been discussion around this issue (Function to specify initial values in brms model has dimension mismatch  - https://discourse.mc-stan.org/t/function-to-specify-initial-values-in-brms-model-has-dimension-mismatch/8692) which resolved only with a method to set one initial value for a specific parameter. However, some models may benefit from setting initial values for some/all parameters and the previous discussion does not elucidate how to do this. There is a previous example for a specific model with the Wiener family (http://singmann.org/wiener-model-analysis-with-brms-part-i/), but this does not generalize to other models (even those that use the Wiener family). Therefore, it remains unclear how to set initial values for specific parameters.

The previous discussion demonstrated that inability to set specific initial values may cause estimation to fail. So, perhaps it would be helpful to many users to have procedures available for doing this.

**Comment by paul-buerkner** [comment]
I see and agree with your point. I will think more of how this can be best
implemented and maintained.

drjhw <notifications@github.com> schrieb am Fr., 24. Apr. 2020, 12:53:

**Comment by drjhw** [comment]
I have found a way to create some sets of fairly-informaive initial values that have the correct structure for brm:-
1) run the model using variational Bayes - e.g. myVBmodel=brm(mybrmsformula, data=mydata, algorithm='meanfield', tol_rel_obj=1e-5, iter=1e5)
2) extract the estimates from the VB model - e.g. myVBmodelEst=myVBmodel$fit@sim$est
3) create a list of the number of sets of estimates that you want to run as chains in your model that uses NUTS sampling - e.g. for a 4-chain NUTS analysis, write: my_inits=list(myVBmodel$fit@sim$est,myVBmodel$fit@sim$est,myVBmodel$fit@sim$est,myVBmodel$fit@sim$est)
5) use the list as the initial values in the analysis that uses NUTS: - e.g. myNUTSmodel=brm(mybrmsformula, data=mydata, chains=4, cores=4, init=my_inits)

I have been able to run some non-linear models that use NUTS sampling using this method - models which would not run if I started them with random initial values. The estimates of the variational Bayes and NUTS sampling models are generally fairly similar (which may be a problem with the method - since it uses the VB estimates as the starting values. But, the standard errors of the variational Bayes models are invariably too small, compared with the NUTS sampling models.

It occurred to me that it would be simpler to estimate the variational Bayes model and then "update" it to use NUTS sampling. But, when I try this, the model begins sampling but then throws errors that the tol_rel_obj is an unknown argument:-

Start sampling
Error in checkForRemoteErrors(val) : 
  4 nodes produced errors; first error: passing unknown argument: tol_rel_obj

I find that it is necessary to reduce tol_rel_obj and use a lot of iterations (1e5 - 1e6) to get reasonable results with algorithm='meanfield'. So, I can't dispense with the tol_rel_obj argument.

Would it be possible to amend update(), so that it will allow the switch from variational Bayes to NUTS sampling?
